src/main/java/org/jetbrains/bsp/bazel/server/BepServer.java

```diff
@@ -88,156 +86,146 @@ public void publishLifecycleEvent(
-          if (event.hasStarted() && event.getStarted().getCommand().equals("build")) {
-            BuildEventStreamProtos.BuildStarted buildStarted = event.getStarted();
-            TaskId taskId = new TaskId(buildStarted.getUuid());
-            TaskStartParams startParams = new TaskStartParams(taskId);
-            startParams.setEventTime(buildStarted.getStartTimeMillis());
-            bspClient.onBuildTaskStart(startParams);
-            taskParkingLot.add(taskId);
-          }
-          if (event.hasFinished()) {
-            BuildEventStreamProtos.BuildFinished buildFinished = event.getFinished();
-            if (taskParkingLot.size() == 0) {
-              System.out.println("No start event id was found.");
-              return;
-            } else if (taskParkingLot.size() > 1) {
-              System.out.println("More than 1 start event was found");
-              return;
-            }
+  public void collectDiagnostics(
+      Map<Uri, List<PublishDiagnosticsParams>> filesToDiagnostics,
+      BuildTargetIdentifier target,
+      String diagnosticsLocation)
+      throws IOException {
+    diagnosticsDispatcher.collectDiagnostics(filesToDiagnostics, target, diagnosticsLocation);
+  }
 
-            TaskFinishParams finishParams =
-                new TaskFinishParams(
-                    taskParkingLot.pop(), convertExitCode(buildFinished.getExitCode().getCode()));
-            finishParams.setEventTime(buildFinished.getFinishTimeMillis());
-            bspClient.onBuildTaskFinish(finishParams);
-          }
+  public void emitDiagnostics(
+      Map<Uri, List<PublishDiagnosticsParams>> filesToDiagnostics, BuildTargetIdentifier target) {
+    diagnosticsDispatcher.emitDiagnostics(filesToDiagnostics, target);
+  }
 
-          if (event.getId().hasNamedSet()) {
-            namedSetsOfFiles.put(event.getId().getNamedSet().getId(), event.getNamedSetOfFiles());
-          }
-          if (event.hasCompleted()) {
-            processCompletionEvent(event);
-          }
-          if (event.hasAction()) {
-            processActionDiagnostics(event);
-          }
-          if (event.hasAborted()) {
-            processAbortion(event.getAborted());
-          }
-          if (event.hasProgress()) {
-            BuildEventStreamProtos.Progress progress = event.getProgress();
-            processStdErrDiagnostics(progress);
-            String message = progress.getStderr().trim();
-            if (!message.isEmpty()) {
-              buildClientLogger.logMessage(progress.getStderr().trim());
-            }
-          }
+  public void handleEvent(BuildEvent buildEvent) {
+    try {
+      processEvent(buildEvent);
+    } catch (IOException e) {
+      LOGGER.error("Error deserializing BEP proto: {}", e.toString());
+    }
+  }
 
-        } catch (IOException e) {
-          System.err.println("Error deserializing BEP proto: " + e);
-        }
-      }
+  void processEvent(BuildEvent buildEvent) throws IOException {
```

Comment:
Imo this method should be deleted (see the comment above), but default visibility is probably not what we want anyway

Reply:
Visibility was an accident, sorry. As to having this extracted method, it was needed when this block in the try-catch was larger but now that the ifs are extracted _processEvent_ is much shorter so maybe you're right about not extracting it.